src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -341,8 +342,19 @@ def _find_or_create_imported_symbol(location, name, scope_limit=None,
         return first_symbol_table.new_symbol(
                 name, interface=UnresolvedInterface(), **kargs)
 
+    # Are there any interfaces that might be hiding the symbol declaration?
+    symbol_table = location.scope.symbol_table
+    try:
```

Comment:
This will always create the symbol in the innermost symbol table. E.g. in the call case I showed in other comments.

Could this be integrated with the "except KeyError" above, so it is created at the first level symbol table where there is a possibility for it to exist?

Also if they are not merged isn't there the possibility with nesting of modules or routines that we create a unresolved symbol in a outer symbol table because it has a unqualified import while the symbol really should come from a inner abstract interface? (tbh I don't know if this is a valid construction in Fortran)

Reply:
I know you suspect that the solution I have added might be more general than visibility symbols but I've been looking and can't find a case. We know that interfaces will hide the declarations of routines. However, due to Fortran's support for object linking a routine can be declared in isolation. Any use of `procedure(abstract_func_name)` for abstract interfaces results in a codeblock. So I think the only case in question is visibility and visibility statements have to be specified within a container (module) which is why I ensure the symbol is creating in place in the local symbol table. I suspect there will be discussion around this so will leave it there.